django/forms/models.py

```diff
@@ -1162,7 +1162,9 @@ def __init__(self, queryset, *, empty_label="---------",
                  required=True, widget=None, label=None, initial=None,
                  help_text='', to_field_name=None, limit_choices_to=None,
                  **kwargs):
-        if required and (initial is not None):
+        #  A blank option is not idiomatic for radio buttons,
+        #  see https://code.djangoproject.com/ticket/26813
+        if required and (initial is not None) or isinstance(widget, RadioSelect):
```

Comment:
Blank option should be removed only if it is not a valid choice. This change removes blank options in all cases. Please check a ticket description.